public/apps/account/tenant-switch-panel.tsx

```diff
@@ -0,0 +1,251 @@
+          />
+        </>
+      ),
+      disabled: customTenantOptions.length === 0,
+    },
+  ];
+
+  const [tenantSwitchRadioIdSelected, setTenantSwitchRadioIdSelected] = useState<string>();
+  const onTenantSwitchRadioChange = (radioId: string) => {
+    setTenantSwitchRadioIdSelected(radioId);
+    setErrorCallOut('');
+  };
+
+  const changeTenant = async (tenantName: string) => {
+    await selectTenant(props.coreStart.http, {
+      tenant: tenantName,
+      username: props.username,
+    });
+  };
+
+  let content;
+
+  if (isMultiTenancyEnabled) {
+    content = (
+      <>
+        <EuiRadioGroup
+          options={tenantSwitchRadios}
+          idSelected={tenantSwitchRadioIdSelected}
+          onChange={(radioId) => onTenantSwitchRadioChange(radioId)}
+          name="tenantSwitchRadios"
+        />
+
+        <EuiSpacer />
+
+        {errorCallOut && (
+          <EuiCallOut color="danger" iconType="alert">
+            {errorCallOut}
+          </EuiCallOut>
+        )}
+      </>
+    );
+  } else {
+    content = <>Contact the administrator to enable multi tenancy</>;
+  }
+
+  return (
+    <EuiOverlayMask>
+      <EuiModal onClose={props.handleClose}>
+        <EuiSpacer />
+        <EuiModalBody>
+          <EuiTitle>
+            <h4>Select your tenant</h4>
+          </EuiTitle>
+
+          <EuiSpacer />
+
+          <EuiText size="s" color="subdued">
+            Tenants are useful for safely sharing your work with other Kibana users. You can switch
+            your tenant anytime by clicking the user avatar on top right.
+          </EuiText>
+
+          <EuiSpacer />
+
+          {content}
+        </EuiModalBody>
+        <EuiModalFooter>
+          <EuiButtonEmpty onClick={props.handleClose}>Cancel</EuiButtonEmpty>
+
+          <EuiButton
+            fill
+            disabled={!isMultiTenancyEnabled}
+            onClick={async function () {
+              let tenantName;
+
+              if (tenantSwitchRadioIdSelected === GLOBAL_TENANT_RADIO_ID) {
+                tenantName = '';
+              } else if (tenantSwitchRadioIdSelected === PRIVATE_TENANT_RADIO_ID) {
+                tenantName = '__user__';
+              } else if (tenantSwitchRadioIdSelected === CUSTOM_TENANT_RADIO_ID) {
+                if (selectedCustomTenantOptions && selectedCustomTenantOptions.length > 0) {
+                  tenantName = comboBoxOptionToString(selectedCustomTenantOptions[0]);
+                }
+              }
+
+              // check tenant name before calling backend
+              if (tenantName === undefined) {
+                setErrorCallOut('No target tenant is specified!');
+              } else {
+                try {
+                  await changeTenant(tenantName);
+                  props.handleClose();
+                } catch (e) {
+                  setErrorCallOut('Failed to switch tenant due to ' + e?.body?.message);
+                }
+              }
+            }}
```

Comment:
For this complex function, I would suggest declare it rather than have it inline.

Reply:
sure.